pkg/sql/catalog/hydrateddesc/hydrate.go

```diff
@@ -68,8 +68,17 @@ func MakeTypeLookupFuncForHydration(
 ) typedesc.TypeLookupFunc {
 	return func(ctx context.Context, id descpb.ID) (tn tree.TypeName, typ catalog.TypeDescriptor, err error) {
 		typDesc := c.LookupDescriptor(id)
-		if typDesc == nil {
-			typDesc, err = lookupFn(ctx, id)
+		// If the type in the fast lookup catalog is a table implicit record type
+		// and  not fully hydrated, we need to make sure we get a hydrated version
+		// of it.
+		// TODO(chengxiong): we may also do topological sort on the descritpors need
+		// to be hydrated and hydrated them in order. In the mean while, we need to
+		// upsert the hydrated version into the fast lookup catalog. This may save
+		// some round trips.
+		if typDesc == nil ||
```

Comment:
@ajwerner I'm not sure how you gonna feel about this change. this is address the case where table descriptor returned from the catalog lookup is not hydrated.

Reply:
I also thought about sorting the descriptors by type, and hydrating them in order. For example, always hydrate type (no hydration needed for it though) and table descriptor first since they're needed to hydrated other descriptors. And we'd need to upsert the hydrated version back to the lookup catalog. However, I don't think this would be much better...